nixos/modules/services/cluster/spark/default.nix

```diff
@@ -0,0 +1,151 @@
+{config, pkgs, lib, ...}:
+let
+  cfg = config.services.spark;
+in
+with lib;
+{
+  options = {
+    services.spark = {
+      master = {
+        enable = mkEnableOption "Enable spark master service";
+        bind = mkOption {
+          type = types.str;
+          description = "Address the spark master binds to";
+          default = "127.0.0.1";
+          example = "0.0.0.0";
+        };
+        restartIfChanged  = mkOption {
+          type = types.bool;
+          description = "Automatically restart master service config change";
+          default = true;
+        };
+        extra-environment = mkOption {
+          type = types.attrsOf types.str;
+          description = "Extra environment variables to pass to spark master. See spark-standalone documentation.";
+          default = {};
+          example = {
+            SPARK_MASTER_WEBUI_PORT = 8181;
+            SPARK_MASTER_OPTS = "-Dspark.deploy.defaultCores=5";
+          };
+        };
+      };
+      worker = {
+        enable = mkEnableOption "Enable spark worker service";
+        workDir = mkOption {
+          type = types.path;
+          description = "Spark worker work dir";
+          default = "/var/lib/spark";
+        };
+        master = mkOption {
+          type = types.str;
+          description = "Address of the spark master";
+          default = "127.0.0.1:7077";
+        };
+        restartIfChanged  = mkOption {
+          type = types.bool;
+          description = "Automatically restart service config change";
+          default = true;
+        };
+        extra-environment = mkOption {
+          type = types.attrsOf types.str;
+          description = "Extra environment variables to pass to spark worker";
+          default = {};
+          example = {
+            SPARK_WORKER_CORES = 5;
+            SPARK_WORKER_MEMORY = "2g";
+          };
+        };
+      };
+      confDir = mkOption {
+        type = types.path;
+        description = "Spark configuration directory";
```

Comment:
I think this`description` needs more explanation.

Reply:
Added more detail from spark docs.